blog/_posts/2025-04-22-Enhance-security-OS-Security-Patch.md

```diff
@@ -0,0 +1,67 @@
+This approach, while effective, introduced challenges. Untested or uncontrolled security packages occasionally caused outages, emphasizing the need for a more reliable and managed patching mechanism at the node level. Additionally, maintaining reboot daemonsets like [kured](https://github.com/kubereboot/kured) added operational overhead for many customers. Ideally, customers prefer managed Kubernetes services, such as Azure Kubernetes Service (AKS), to handle OS security patching comprehensively and seamlessly.
+
+## Automatic Node OS Security Patching mechanisms at AKS
+
+AKS provides two managed and tested mechanisms to deliver the latest security packages to your Node Operating System. 
+
+**Automatic Node Image Channel** - AKS updates nodes weekly with a newly patched VHD for security and bug fixes. This update follows maintenance windows and upgrade configuration settings. Automatic Node image upgrades are supported as long as the cluster's Kubernetes minor version is in support. These AKS-tested node images are fully managed and applied with safe deployment practices.
+
+**OS Security Patch Channel** - Several customers may need only the security packages for their OS without additional bug fixes and updates. The OS Security Patch channel provides a fully managed, attended Node OS security-only solution. The Security-Patch channel reimages nodes only when necessary and provides live security patching updates with zero disruption, respecting planned maintenance windows and follows azure safe deployment practices.
+
+
+## When to use OS Security Patch or Automatic Node Image Channel?
+
+Choosing between the OS Security Patch channel and the Automatic Node Image channel depends on your specific requirements and operational constraints. Here's a breakdown based on common scenarios:
+
+- **Speed of Patching is Paramount**: If addressing critical vulnerabilities (CVEs) promptly is a priority, the OS Security Patch channel is the better choice. It delivers updates 1 to 2 weeks faster than the Node Image channel, ensuring your workloads remain secure against emerging threats.
+
+- **Require Comprehensive Security Fixes and Bug Fixes**: For environments where both security patches and additional bug fixes or binaries are essential, the Node Image channel is ideal. It provides a more comprehensive update approach, ensuring both security and functionality improvements.
+
+- **Workload Sensitive to Multiple Reimages in a Month**: If your workloads cannot tolerate frequent disruptions, the OS Security Patch channel is preferable. It minimizes disruptions by focusing solely on security packages, reimaging nodes 60-70% less frequently, and performing live security patching during other times.
+
+- **Using Windows Environment for Running Workloads**: Currently, the OS Security Patch channel is better suited for Linux-based workloads. For Windows environments, the Node Image channel is recommended until OS Security Patch support for Windows is introduced.
```

Comment:
Better suited? Does it work with windows? Flourished language doesn't really help in technical docs, if it's not supported yet, then it's not supported yet. Also Az Linux is fully supported by the time this blog is out?

Reply:
Az linux is already supported except it does not have the Snapshoting of Ubuntu and therefore yet to integrate into all our tests and rel tracker.